File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
     
        if(s == null || s.length() ==0)return -1;
        Map<Character, Integer> countMap = new HashMap<>();
        for(char c: s.toCharArray()){
            countMap.put(c, countMap.getOrDefault(c,0)+1);
        }
        for(int i=0;i< s.length(); i++) {
            if(countMap.get(s.charAt(i)) ==1)return i;
        }
        return -1;
    }
}
//time and space complexity:O(n)

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        #check for null input
        if not s: return -1
        hmap = {}
        for c in s: 
            if c not in hmap: hmap[c]=1
            else: hmap[c]+=1
        for i in range(len(s)):
            if hmap[s[i]] ==1: return i
        return -1
#time and space compelxity: O(n)